### PR TITLE
feat: Generate auto-filled config

### PR DESCRIPTION
**Pull request by hermanschaaf**
Follow-up of https://github.com/cloudquery/cloudquery/pull/1698 but merging to main branch.

### CODE REVIEW

**Review by erezrokah — APPROVED** [review]

**Comment by hermanschaaf** [comment]
I'm going to try and move most of these changes to the SDK, and then refactor this PR. Let's hold off on merging

**Comment by hermanschaaf** [comment]
This branch now depends on the changes in https://github.com/cloudquery/plugin-sdk/pull/61 and the go.mod will need to be updated here first to bring in those changes once they are merged in, before we can merge this branch.

**Pull request closed without merging**
